parser.go

```diff
@@ -64,7 +64,11 @@ func (c *cache) getValue() *Value {
 	if cap(c.vs) > len(c.vs) {
 		c.vs = c.vs[:len(c.vs)+1]
 	} else {
-		c.vs = append(c.vs, Value{})
+		if len(c.vs) == 0 {
```

Comment:
`append` already does something similar underneath - see https://play.golang.org/p/wFzqBZvEXPl . If `append` works slower than this code, then a bug must be filled at https://github.com/golang/go/issues